rp2040-hal/src/clocks/mod.rs

```diff
@@ -301,6 +305,46 @@ impl ClocksManager {
             .configure_clock(&self.system_clock, self.system_clock.freq())
     }
 
+    /// Approximates the frequency of the given clock source.
+    pub fn approximate_frequency<C: ClockSource>(&mut self, _trg_clk: C) -> HertzU32 {
+        // Wait for the frequency counter to be ready
+        while self.clocks.fc0_status.read().running().bit_is_set() {
+            core::hint::spin_loop()
+        }
+
+        // Set the speed of the reference clock in kHz.
+        self.clocks.fc0_ref_khz.write(|w| unsafe {
+            w.fc0_ref_khz()
+                .bits(self.reference_clock.get_freq().to_kHz())
```

Comment:
It's not obvious if this is safe: `fc0_ref_khz`  only has 20 non-reserved bits, so writing a larger value might be undefined.